File: DataStructure/leetcode525/solveLeetcode525.go
// func findMaxLength(nums []int) int {
//     n := len(nums)
//     hash := make(map[int]int)
//     hash[0] = -1

//     prefix := 0
//     ans := 0
//     for i := 0; i < n; i++ {
//         prefix += (nums[i] * 2) - 1

//         if val, ok := hash[prefix]; ok {
//             if ans < i-val {
//                 ans = i - val
//             }
//         } else {
//             hash[prefix] = i
//         }
//     }

//     return ans
// }

File: DataStructure/leetcode525/solveLeetcode525.cpp
// class Solution 
// {
// public:
//     int findMaxLength(vector<int>& nums) 
//     {
//         const int n=nums.size();
//         unordered_map<int,int>hash;
//         hash[0]=-1;

//         int prefix=0,ans=0;
//         for(int i=0;i<n;++i)
//         {
//             prefix+=(nums[i]?1:-1);

//             if(hash.count(prefix))
//                 ans=max(ans,i-hash[prefix]);
//             else hash[prefix]=i;
//         }

//         return ans;
//     }
// };